commands/lyrics.js

```diff
@@ -1,39 +1,164 @@
+    await run(client, channel, send, member)
 }
 
-command.prefixRun = async function prefixRun(client, message, parameters)
-{
+command.prefixRun = async function prefixRun(client, message) {
     const channel = message.channel
+    const member = message.member
     const send = channel.send.bind(channel)
-    
-    await channel.send('Lets ask Kermit!')
 
-    await run(client, channel, send)
+    await run(client, channel, send, member)
 }
 
-async function run(client, channel, send)
-{
+async function run(client, channel, send, member) {
+
+    const lyricsToken = client.lyricsToken
+    const geniusClient = new Genius.Client(lyricsToken)
     const guildId = channel.guild.id
+    const queue =  client.player.getQueue(guildId)
+
+    if (!queue || !queue.playing) return send('No music is currently playing.')
+
+    let currentPage = 0
+    const songList = await getSongList(geniusClient, userSongs, channel, queue)
+    const songLyrics = await getLyrics(songList, send, currentPage)
+    const lyricEmbed = await createLyricsEmbed(client, songLyrics, songList, currentPage)
+    const lyricButtons = await createLyricsButtons(client, songList, currentPage)
+    const message = await send({ embeds: [lyricEmbed], components: [lyricButtons] })
+    
+    handlePagination(client, message, member, songList, send, currentPage)
+}
+
+async function getSongList(lyrics, songs, channel, queue) {
+
+    const userSongName = songs[0]
+    const currentQueueSongName = queue.songs[0].name
+    const similaritiesRate = await compareSongsName(userSongName, currentQueueSongName)
+    const similaritiesThreshold = 0.1
+
+    if( similaritiesRate >= similaritiesThreshold) {
+        const userSongList = await lyrics.songs.search(userSongName)
+        const queueSongList = await lyrics.songs.search(currentQueueSongName)
+
+        return userSongList.length === 0 ? queueSongList : userSongList
+    } else {
+        const queueSongList =  await lyrics.songs.search(currentQueueSongName)
+        
+        return queueSongList
+    }
+
+}
+
+async function getLyrics(songList, send, currentPage) {
+
+    const firstSong = songList[currentPage]
+    if (firstSong === undefined) return send('No lyrics found for this song')
+
+    const currentSongLyrics = await firstSong.lyrics();
+
+    return currentSongLyrics
+}
+
+async function compareSongsName(userSongName, queueSongName){
+    const userSong = userSongName.split(' ')
+    const queueSong = queueSongName.split(' ')
+    const intersection = userSong.filter(songName => queueSong.includes(songName))
+    const similaritiesValue = intersection.length / Math.max(userSong.length, queueSong.length)
+
+    return similaritiesValue
+}
+
+async function handlePagination(client, message, member, songList, send, currentPage){
+    const commandAuthorFilter = i => i.user.id === member.id
+
+    const collector = message.createMessageComponentCollector({ commandAuthorFilter, time: 300000 })
```

Comment:
Could you add a comment that says how many second this is please?

Reply:
Added a comment and turned it into a constant instead of just a magic number